php/php.editor/src/org/netbeans/modules/php/editor/codegen/CGSInfo.java

```diff
@@ -369,17 +395,40 @@ private String getFirstTypeFromBlock(final PHPDocBlock phpDoc) {
         }
 
         private String getFirstTypeFromTag(final PHPDocTypeTag typeTag) {
+            boolean canBeNull = canBeNull(typeTag);
             String result = ""; //NOI18N
             for (PHPDocTypeNode typeNode : typeTag.getTypes()) {
                 String type = typeNode.getValue();
-                if (!Type.isPrimitive(type) && !VariousUtils.isSpecialClassName(type)) {
+                if (phpVersion.compareTo(PhpVersion.PHP_70) >= 0
+                        && !VariousUtils.isSpecialClassName(type)
+                        && !Type.isInvalidPropertyType(type)) {
+                    result = typeNode.isArray() ? Type.ARRAY : type;
+                    if (canBeNull && phpVersion.compareTo(PhpVersion.PHP_71) >= 0) {
+                        result = CodeUtils.NULLABLE_TYPE_PREFIX + result;
+                    }
+                    break;
+                } else if (!Type.isPrimitive(type) && !VariousUtils.isSpecialClassName(type)) {
                     result = typeNode.isArray() ? Type.ARRAY : type;
                     break;
                 }
             }
             return result;
         }
 
+        private boolean canBeNull(final PHPDocTypeTag typeTag) {
+            boolean canBeNull = false;
+            if (typeTag.getTypes().size() > 1) {
+                for (PHPDocTypeNode typeNode : typeTag.getTypes()) {
+                    String type = typeNode.getValue().toLowerCase();
```

Comment:
`toLowerCase()` should be always called with a parameter (locale), `EN_US` should work fine in this case.